Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule.py

```diff
@@ -1572,6 +1572,72 @@ def remove_user_role(self, user_emails: list[str]) -> dict[str, dict[str, str]]:
             }},
         )
 
+    def terminate_process(self, agent_id, instance_id, process_name, incident_id) -> dict[str, dict[str, str]]:
+        """
+            Terminate a specific process on an agent.
+
+            :type agent_id: ``str``
+            :param agent_id: The ID of the agent.
+
+            :type instance_id: ``str``
+            :param instance_id: The ID of the instance.
+
+            :type process_name: ``Optional[str]``
+            :param process_name: The name of the process. Optional.
+
+            :type incident_id: ``Optional[str]``
+            :param incident_id: The ID of the incident. Optional.
+
+            :return: The response from the API.
+            :rtype: ``dict[str, dict[str, str]]``
+        """
+        request_data: Dict[str, Any] = {
+            "agent_id": agent_id,
+            "instance_id": instance_id,
+        }
+        if process_name:
+            request_data["process_name"] = process_name
+        if incident_id:
+            request_data["incident_id"] = incident_id
+        return self._http_request(
+            method='POST',
+            url_suffix="/endpoints/terminate_process/",
+            json_data={"request_data": request_data},
+        )
+
+    def terminate_causality(self, agent_id: str, causality_id: str, process_name: Optional[str], incident_id: Optional[str]) -> dict[str, dict[str, str]]:
```

Comment:
Those methods are practically the same. Let's think of a smart way to reduce code duplication